src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/AmazonSqsReceiver.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\AmazonSqs\Transport;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\LogicException;
+use Symfony\Component\Messenger\Exception\MessageDecodingFailedException;
+use Symfony\Component\Messenger\Exception\TransportException;
+use Symfony\Component\Messenger\Transport\Receiver\MessageCountAwareInterface;
+use Symfony\Component\Messenger\Transport\Receiver\ReceiverInterface;
+use Symfony\Component\Messenger\Transport\Serialization\PhpSerializer;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+use Symfony\Contracts\HttpClient\Exception\HttpExceptionInterface;
+
+/**
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class AmazonSqsReceiver implements ReceiverInterface, MessageCountAwareInterface
+{
+    private $connection;
+    private $serializer;
+
+    public function __construct(Connection $connection, SerializerInterface $serializer = null)
+    {
+        $this->connection = $connection;
+        $this->serializer = $serializer ?? new PhpSerializer();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get(): iterable
+    {
+        try {
+            $sqsEnvelope = $this->connection->get();
+        } catch (HttpExceptionInterface $e) {
+            throw new TransportException($e->getMessage(), 0, $e);
+        }
+        if (null === $sqsEnvelope) {
+            return;
+        }
+
+        try {
+            $envelope = $this->serializer->decode([
+                'body' => $sqsEnvelope['body'],
+                'headers' => $sqsEnvelope['headers'],
+            ]);
+        } catch (MessageDecodingFailedException $exception) {
+            $this->connection->delete($sqsEnvelope['id']);
```

Comment:
Shouldn't the message be rejected?

Reply:
Reject and delete is the same in SQS context.

Reply:
Correct correct.. 
As we know you can configure the queue from aws after x retries to move it to "dead letter queue"..

what I meant is that this will not move the message to the dead letter queue.. or am I wrong? shouldn't we somehow be able either to manually move them to dead letter if symfony is configured to do so.. or if the developer has configured the queue to automatically move it after X retries to notify the queue to move it?

> or am I wrong

If I am wrong then forget about.. I might miss something here..

Reply:
`symfony/messenger` use a custom abstraction to handle retries and failures (https://symfony.com/doc/current/messenger.html#saving-retrying-failed-messages)

Failed messages are not retried from a "backend" point of view. A new message is created a the previous one is rejected (see code of SendFailedMessageForRetryListener and SendFailedMessageToFailureTransportListener)
DeadLetter is also managed by symfony by creating a new Message in another queue/backend.

Keeping "backend" behavior alongside Symfony custom logic will result in duplicate messages.

Reply:
ok I guess we can resolve this then